astropy/coordinates/builtin_frames/galactocentric.py

```
@@ -88,76 +98,95 @@ class galactocentric_frame_defaults(ScienceState):
-            references['galcen_distance'] = \
-                'https://ui.adsabs.harvard.edu/#abs/2009ApJ...692.1075G'
-
-            params['galcen_v_sun'] = r.CartesianDifferential([11.1,
-                                                              220+12.24,
-                                                              7.25]*u.km/u.s)
-            references['galcen_v_sun'] = \
-                ['https://ui.adsabs.harvard.edu/#abs/2010MNRAS.403.1829S',
-                 'https://ui.adsabs.harvard.edu/#abs/2015ApJS..216...29B']
-
-            params['z_sun'] = 27.0 * u.pc
-            references['z_sun'] = \
-                'https://ui.adsabs.harvard.edu/#abs/2001ApJ...553..184C'
-
-        elif arg == 'v4.0':
-            params['galcen_distance'] = 8.122 * u.kpc
-            references['galcen_distance'] = \
-                'https://ui.adsabs.harvard.edu/abs/2018A%26A...615L..15G'
-
-            params['galcen_v_sun'] = r.CartesianDifferential([12.9,
-                                                              245.6,
-                                                              7.78]*u.km/u.s)
-            references['galcen_v_sun'] = \
-                ['https://ui.adsabs.harvard.edu/abs/2018RNAAS...2..210D',
-                 'https://ui.adsabs.harvard.edu/abs/2018A%26A...615L..15G',
-                 'https://ui.adsabs.harvard.edu/abs/2004ApJ...616..872R']
-
-            params['z_sun'] = 20.8 * u.pc
-            references['z_sun'] = \
-                'https://ui.adsabs.harvard.edu/abs/2019MNRAS.482.1417B'
-
+        if arg in cls._registry:
+            state = cls._registry[arg].copy()
+            params = dict(state.get("value"))  # ensure mutable
+            references = dict(state.get("references", dict()))
         else:
-            raise ValueError(f'Invalid string input to retrieve solar '
-                             f'parameters for Galactocentric frame: "{arg}"')
+            raise ValueError(
```

Hmm, does feel like it should be a `KeyError`... In fact, as we're changing API here anyway by returning an extra `state`, I'd be tempted to remove the whole clause and just let the indexing fail if there's no match.

---

I agree. I didn't change things like this so that this PR had the same I/O was before.
I'll change it now.